### DISCUSSION TITLE
Cannot upgrade from 14.1.0 minimal on a NOUS A1T

### DISCUSSION THREAD

**Discussion by Cruc1bl3** [arendst/Tasmota#21587] [Q&A]
### PROBLEM DESCRIPTION
Device:  NOUS A1T Smart Socket with power monitoring
Original Verion: 13.0.0
Original features enabled: MQTT, Syslog
Chipset: 8266, 1M Flash
Upgrade path: 13.0.0-sensors -> 14.1.0 minimal -> 14.1.0 sensors (hopefully)

I have a lot of these smart sockets and the majority updated correctly apart from two; one is bricked and requires flashing via serial port and the other refuses to upgrade beyond minimal.

I've tried the upgrade via the web interface and from the console:

`OtaUrl http://ota.tasmota.com/tasmota/release/tasmota-sensors.bin.gz`
`Upgrade 1`

It says it is upgrading and will reset but after reset it is still minimal. Upgrading via file upload does not work because of inadequate space.

`Jun  7 11:50:45 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/stat/tasmotaXXXXXX/RESULT = {"OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz"}`
`Jun  7 11:51:18 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/tele/tasmotaXXXXXX/STATE = {"Time":"2024-06-07T10:51:18","Uptime":"0T00:20:10","UptimeSec":1210,"Vcc":3.667,"Heap":33,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"XXXXXXX","BSSId":"AA:BB:CC:DD:EE:FF","Channel":6,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:04"}}
`
`Jun  7 11:51:24 192.168.200.200 tasmota-XXXXXX ESP-CMD: otaurl http://ota.tasmota.com/tasmota/release/tasmota-sensors.bin.gz
`
`Jun  7 11:51:24 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/stat/tasmotaXXXXXX/RESULT = {"OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-sensors.bin.gz"}
`
`Jun  7 11:51:39 192.168.200.200 tasmota-XXXXXX ESP-CMD: upgrade 1
`
`Jun  7 11:51:39 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/stat/tasmotaXXXXXX/RESULT = {"Upgrade":"Version 14.1.0 from http://ota.tasmota.com/tasmota/release/tasmota-sensors.bin.gz"}
`
`Jun  7 11:52:34 192.168.200.200 tasmota-XXXXXX ESP-MQT: Attempting connection...
`
`Jun  7 11:52:34 192.168.200.200 tasmota-XXXXXX ESP-MQT: Connected
`
`Jun  7 11:52:34 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/tele/tasmotaXXXXXX/LWT = Online (retained)
`
`Jun  7 11:52:34 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/cmnd/tasmotaXXXXXX/POWER = 
`
`Jun  7 11:52:34 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/stat/tasmotaXXXXXX/RESULT = {"Command":"Unknown"}
`
`Jun  7 11:55:14 192.168.200.200 tasmota-XXXXXX ESP-CFG: Loaded from flash at FB, Count 192
`
`Jun  7 11:55:14 192.168.200.200 tasmota-XXXXXX ESP-Project tasmota - sonoff11 Version 14.1.0(release-minimal)-2_7_6(2024-06-03T11:53:06)
`
`Jun  7 11:55:14 192.168.200.200 tasmota-XXXXXX ESP-WARNING This version does not support persistent settings
`
`Jun  7 11:55:14 192.168.200.200 tasmota-XXXXXX ESP-WIF: Connecting to AP1 YYYYYYY Channel 6 BSSId AA:BB:CC:DD:EE:FF in mode 11n as tasmota-XXXXXX...
`
`Jun  7 11:55:14 192.168.200.200 tasmota-XXXXXX ESP-WIF: Connected
`
`Jun  7 11:55:14 192.168.200.200 tasmota-XXXXXX ESP-HTP: Web server active on tasmota-XXXXXX with IP address 192.168.200.200
`
`Jun  7 11:55:15 192.168.200.200 tasmota-XXXXXX ESP-MQT: Attempting connection...
`
`Jun  7 11:55:15 192.168.200.200 tasmota-XXXXXX ESP-MQT: Connected
`
`Jun  7 11:55:15 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/tele/tasmotaXXXXXX/LWT = Online (retained)
`
`Jun  7 11:55:15 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/cmnd/tasmotaXXXXXX/POWER = 
`
`Jun  7 11:55:15 192.168.200.200 tasmota-XXXXXX ESP-MQT: pdu/tele/tasmotaXXXXXX/INFO1 = {"Info1":{"Module":"NOUS A1T","Version":"14.1.0(release-minimal)","FallbackTopic":"cmnd/DVESXXXXXX_fb/","GroupTopic":"pdu/cmnd/tasmotas/"}}`

What is curious is that it has retained the syslog and MQTT settings, I'm still getting these messages evn though there isn't a config option for it.

Does anyone have any advice before I dismantle the socket and flash them via the serial port?

Thanks,

Simon

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: _____
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
13:26:39.117 MQT: pdu/stat/tasmota_154878/RESULT = {"Command":"Unknown"}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
13:26:39.117 MQT: pdu/stat/tasmota_154878/RESULT = {"Command":"Unknown"}

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
13:26:39.117 MQT: pdu/stat/tasmota_154878/RESULT = {"Command":"Unknown"}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
13:28:21.976 MQT: pdu/stat/tasmota_154878/RESULT = {"WebLog":4}

```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

A nice tear down of this socket can be seen here [https://nathaliebeimler.com/tech/Nous_A1.html](url)
**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Noschvie** [comment]
Try tasmota.bin.gz, do you really need the sensors version?

**Reply by Cruc1bl3** [reply]
Yes, the sensors version contains the power monitoring and display code.

**Reply by Noschvie** [reply]
I'm running these devices with tasmota.bin and it is working well.

**Comment by Jason2866** [comment]
Do not connect any sensors to power monitoring devices. It is dangerous! GPIOs have AC mains level.
Do the upgrade with the tasmota firmware.

**Reply by Cruc1bl3** [reply]
This is a commercial product, not something home grown, the power sensing is provided out of the box. For more info see https://www.amazon.de/s?k=nous+a1t&crid=24KK6GS4B7LE0&sprefix=nous+a1t%2Caps%2C88&ref=nb_sb_noss_1

**Reply by sfromis** [reply]
I don't see *anything* to suggest that Nous A1T has the energy monitoring "out of the box". Are you talking about a completely different device than that? 
![image](https://github.com/arendst/Tasmota/assets/47082390/c166069a-33c2-4539-9ca3-941fd024b7a8)
(A generic search on a site known for its horrible searching does not tell any details).

**Comment by Noschvie** [comment]
![grafik](https://github.com/arendst/Tasmota/assets/48217146/235b2402-6b73-4494-9b13-8ac608f9d671)

**Comment by Jason2866** [comment]
As said power monitoring smart plug. Sensor firmware is the wrong choice for.

**Reply by sfromis** [reply]
Yup, standard `tasmota.bin.gz`, just don't use the "lite" firmware, which does not include energy monitoring.

**Comment by sfromis** [comment]
Two details worthy of note:

The log shown at that start does not have **any** symptoms of errors, just what looks very normal, but cut off before it becomes "interesting".

And after a successful upgrade, the Tasmota web UI will still show a big red "MINIMAL" at the top, until you refresh the page.

**Comment by Cruc1bl3** [comment]
You are correct, the standard "tasmota" version does work correctly and displays the energy parameters.

Further observations.  This only seems to happen in my testing if the wifi signal is weak and/or noisy for the device (it has a small attennae). I ran a sniffer on the wireless link and what seems to happen is that the OTA firmware download fails part way through but is not logged by the device, after which it reboots automatically. I guess it rejects the incomplete firmware.  It suggests that failure isn't trapped and reported by the download routine.


The bricked unit was recovered by a connecting a programmer to the rx/tx. vcc and gnd pins which are exposied on the underside of the board.
